server/src/main/java/org/candlepin/async/tasks/EntitleByProductsJob.java

```diff
@@ -0,0 +1,183 @@
+
+            final List<Entitlement> ents = entitler.bindByProducts(
+                prodIds, consumerUuid, entitleDate, Arrays.asList(pools));
+            entitler.sendEvents(ents);
+            return "Entitlements created for owner";
+        }
+        // Catch any exception that is fired and re-throw as a JobExecutionException
+        // so that the job will be properly cleaned up on failure.
+        catch (Exception e) {
+            log.error("EntitlerJob encountered a problem.", e);
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    public static EntitleByProductsJobConfig createConfig() {
+        return new EntitleByProductsJobConfig();
+    }
+
+    /**
+     * Job configuration object for the entitle by products job
+     */
+    public static class EntitleByProductsJobConfig extends JobConfig {
+
+        public EntitleByProductsJobConfig() {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME)
+                .addConstraint(JobConstraints.uniqueByArguments(OWNER_KEY));
+        }
+
+        public EntitleByProductsJobConfig setOwner(final Owner owner) {
+            if (owner == null) {
+                throw new IllegalArgumentException("Owner is null");
+            }
+
+            this.setJobMetadata(OWNER_KEY, owner.getKey())
+                .setJobArgument(OWNER_KEY, owner.getKey())
+                .setLogLevel(owner.getLogLevel());
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setConsumer(final Consumer consumer) {
+            if (consumer == null) {
+                throw new IllegalArgumentException("Consumer is null");
+            }
+            this.setJobArgument(CONSUMER_UUID_KEY, consumer.getUuid());
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setProductIds(final String[] prodIds) {
+            if (prodIds == null) {
+                throw new IllegalArgumentException("Product ids is null");
+            }
+            this.setJobArgument(PROD_IDS_KEY, prodIds);
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setEntitleDate(final Date entitleDate) {
+            this.setJobArgument(ENTITLE_DATE_KEY, entitleDate);
+
+            return this;
+        }
+
+        public EntitleByProductsJobConfig setPools(final Collection<String> pools) {
+            if (pools == null) {
+                throw new IllegalArgumentException("From pools is null");
+            }
+            this.setJobArgument(FROM_POOLS_KEY, pools.toArray());
+
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                final JobArguments arguments = this.getJobArguments();
+
+                final String ownerKey = arguments.getAsString(OWNER_KEY);
```

Comment:
The owner key is not an argument of this job's execution, so we don't need to validate its existence here.

Reply:
Done.